docs/FilePropertyManagement.md

```diff
@@ -0,0 +1,150 @@
+
+
+
+## Default properties
+
+Default properties can be set in the corresponding language properties file.  For example, the COBOL file properties can be set in [application-conf/Cobol.properties](../samples/application-conf/Cobol.properties), while the Assembler file properties can be set in [application-conf/Assembler.properties](../samples/application-conf/Assembler.properties), and so on.
+
+zAppBuild is currently proposing to store these properties within the application repository and let the application team have control over these files. If you are looking for a more centralized way to manage the default options for all applications, you can move these definitions into the zAppBuild build framework itself by either merging them into the appropriate language property file under [build-conf](../build-conf/) or store them in a separate directory within zAppBuild itself and leverage the `applicationConfRootDir` property in [build-conf/build.propertiees](../build-conf/build.properties).
+
+## Overriding properties
+
+The following section describes the various strategies to override the default value. The DBB file property syntax is the most commonly used approach within the zAppBuild samples. Two alternate approaches to override build properties are implemented in zAppBuild to serve the different needs and requirement to simplify the adoption of zAppBuild by either leveraging an individual properties file per application artifact or by defining a language definition mapping.
+
+### DBB file properties
+
+The most common way to override a single build parameter, makes use of the DBB file property syntax.
+
+For example, to use a DBB file property to override the default COBOL compile parameters for build artifact `MortgageApplication/cobol/epsnbrvl.cbl`, follow the DBB file property syntax in `file.properties`, which is the default property file for configuring file property overrides:
+
+```properties
+cobol_compileParms=LIB,SOURCE :: **/cobol/epsnbrvl.cbl
+```
+
+For merging, rather than overriding, the property value of this file-level override with the default setting, you can specify the following syntax:
+
+```properties
+cobol_compileParms=${cobol_compileParms},SOURCE :: **/cobol/epsnbrvl.cbl
+```
+
+The file property path syntax also allows you to override the a build parameter for a set of files using wildcards. For example, let's assume that you are storing all CICS modules in `cobol_cics` subfolder. Using the following sample will ensure that the file flag `isCICS` is set to `true` for all COBOL files in this subfolder with the file extension `*.cbl`. However, it is recommended not to store information about the build configuration within the layout of folders, because an update would require to move files into different directories.
```

Comment:
> because an update would require to move files into different directories.

Just to clarify, are we trying to say that we don't recommend storing build config information within subfolders because an update might require the files to be moved into different directories?

Reply:
Hi Lauren, yes. This is correct. I rather prefer to use the folder layout to represent the functional context of the files.

Also, when moving a file to a different subfolder, dbb-zappbuild will recognise it as a changed file and process it. (which is ok).

Maybe we should reword it to:

```
However, it is not recommended to store information about the build configuration within the layout of folders. In that case, updating the build information of a file requires to move it into different directories. We rather prefer to use the folder layout to represent the functional context of the files.
```

Reply:
Would wording it in the following way still be accurate?:
> **Note:** We do not recommend organizing the layout of repository folders/files based on build property management, because future updates to the build information of a file could require it to be moved into different folders. Instead, we recommend that the repository's folder layout represent the functional context of the files.

Reply:
Confirmed with @dennis-behm